### PR TITLE
Event source concept

### PR DESCRIPTION
**Pull request by woksin**
## Summary

Update to new contracts that changes partition and event source id to be a string instead of a guid. Fix all build errors that

### CODE REVIEW

File: Source/events/EventStore.ts [hunk 1]
```diff
@@ -193,7 +193,7 @@ export class EventStore extends IEventStore {
         return new CommittedAggregateEvents(eventSourceId, aggregateRootId, ...convertedEvents);
     }
 
-    private toUncommittedEvent(content: any, eventSourceId: EventSourceId | Guid | string, eventTypeOrId?: EventType | EventTypeId | Guid | string, isPublic = false): UncommittedEvent {
+    private toUncommittedEvent(content: any, eventSourceId: EventSourceId | string, eventTypeOrId?: EventType | EventTypeId | Guid | string, isPublic = false): UncommittedEvent {
```
Comment (jakhog) [inline Source/events/EventStore.ts:196]: Let's bring the Guid conversions back until we get the poll data :)

File: Source/eventHorizon/SubscriptionBuilderForProducerStream.ts [hunk 2]
```diff
@@ -33,9 +33,9 @@ export class SubscriptionBuilderForProducerStream {
 
     /**
      * Sets the producer stream to subscribe to events from.
-     * @param {PartitionId | Guid | string} tenant Stream to subscribe to events from.
+     * @param {PartitionId | string} partitionId Stream partition to subscribe to events from.
      */
-    fromProducerPartition(partitionId: PartitionId | Guid | string): SubscriptionBuilderForProducerPartition {
+    fromProducerPartition(partitionId: PartitionId | string): SubscriptionBuilderForProducerPartition {
```
Comment (jakhog) [inline Source/eventHorizon/SubscriptionBuilderForProducerStream.ts:38]: Guid here

File: Source/events.filtering/PartitionedFilterResult.ts [hunk 3]
```diff
@@ -13,9 +13,9 @@ export class PartitionedFilterResult {
     /**
      * Initializes a new instance of {@link PartitionedFilterResult}.
      * @param {boolean} shouldInclude Tells whether or not the event should be included.
-     * @param {PartitionId | Guid | string} partitionId Tells which partition the event should be partitioned into.
+     * @param {PartitionId | string} partitionId Tells which partition the event should be partitioned into.
```
Comment (jakhog) [inline Source/events.filtering/PartitionedFilterResult.ts:16]: Guid here

File: Source/events/EventSourceId.ts [hunk 4]
```diff
@@ -31,8 +31,8 @@ export class EventSourceId extends ConceptAs<Guid, '@dolittle/sdk.events.EventSo
      * @param {EventSourceId | Guid | string)} id
      * @returns {EventSourceId}
      */
-    static from(id: EventSourceId | Guid | string): EventSourceId {
+    static from(id: EventSourceId | string): EventSourceId {
```
Comment (jakhog) [inline Source/events/EventSourceId.ts:34]: Guid here

File: Source/events/EventStore.ts [hunk 5]
```diff
@@ -58,31 +58,31 @@ export class EventStore extends IEventStore {
     }
 
     /** @inheritdoc */
-    commit(event: any, eventSourceId: EventSourceId | Guid | string, eventType?: EventType | EventTypeId | Guid | string, cancellation?: Cancellation): Promise<CommitEventsResult>;
+    commit(event: any, eventSourceId: EventSourceId | string, eventType?: EventType | EventTypeId | Guid | string, cancellation?: Cancellation): Promise<CommitEventsResult>;
     commit(eventOrEvents: UncommittedEvent | UncommittedEvent[], cancellation?: Cancellation): Promise<CommitEventsResult>;
-    commit(eventOrEvents: any, eventSourceIdOrCancellation?: EventSourceId | Guid | string | Cancellation, eventType?: EventType | EventTypeId | Guid | string, cancellation?: Cancellation): Promise<CommitEventsResult> {
+    commit(eventOrEvents: any, eventSourceIdOrCancellation?: EventSourceId | string | Cancellation, eventType?: EventType | EventTypeId | Guid | string, cancellation?: Cancellation): Promise<CommitEventsResult> {
```
Comment (jakhog) [inline Source/events/EventStore.ts:63]: Guid here

File: Source/events/EventStore.ts [hunk 6]
```diff
@@ -58,31 +58,31 @@ export class EventStore extends IEventStore {
...
             return this.commitInternal(this.asArray(eventOrEvents), eventSourceIdOrCancellation as Cancellation);
         }
-        const eventSourceId = eventSourceIdOrCancellation as Guid | string;
+        const eventSourceId = eventSourceIdOrCancellation as string;
         return this.commitInternal([this.toUncommittedEvent(eventOrEvents, eventSourceId, eventType, false)], cancellation);
     }
 
     /** @inheritdoc */
-    commitPublic(event: any, eventSourceId: EventSourceId | Guid | string, eventType?: EventType | EventTypeId | Guid | string, cancellation?: Cancellation): Promise<CommitEventsResult> {
+    commitPublic(event: any, eventSourceId: EventSourceId | string, eventType?: EventType | EventTypeId | Guid | string, cancellation?: Cancellation): Promise<CommitEventsResult> {
```
Comment (jakhog) [inline Source/events/EventStore.ts:72]: Guid here

File: Source/events/EventStore.ts [hunk 7]
```diff
@@ -58,31 +58,31 @@ export class EventStore extends IEventStore {
...
+    commitForAggregate(event: any, eventSourceId: EventSourceId | string, aggregateRootId: AggregateRootId, expectedAggregateRootVersion: AggregateRootVersion, eventType?: EventType | EventTypeId | Guid | string, cancellation?: Cancellation): Promise<CommitAggregateEventsResult>;
     commitForAggregate(events: UncommittedAggregateEvents, cancellation?: Cancellation): Promise<CommitAggregateEventsResult>;
-    commitForAggregate(eventOrEvents: any, eventSourceIdOrCancellation?: EventSourceId | Guid | string | Cancellation, aggregateRootId?: AggregateRootId, expectedAggregateRootVersion?: AggregateRootVersion, eventType?: EventType | EventTypeId | Guid | string, cancellation?: Cancellation): Promise<CommitAggregateEventsResult> {
+    commitForAggregate(eventOrEvents: any, eventSourceIdOrCancellation?: EventSourceId | string | Cancellation, aggregateRootId?: AggregateRootId, expectedAggregateRootVersion?: AggregateRootVersion, eventType?: EventType | EventTypeId | Guid | string, cancellation?: Cancellation): Promise<CommitAggregateEventsResult> {
```
Comment (jakhog) [inline Source/events/EventStore.ts:81]: Guid here

**Review by jakhog — COMMENTED** [review]
I have commented all the places where the Guid should come back if it needs to

**Comment by woksin** [comment]
Merging this in as I've said that I will fix the SDK that is in the wrong :)

**Pull request merged**